MNE/rtProcessing/rthpis.cpp

```diff
@@ -79,113 +79,93 @@ using namespace INVERSELIB;
 // DEFINE MEMBER METHODS
 //=============================================================================================================
 
-RtHPIS::RtHPIS(FiffInfo::SPtr p_pFiffInfo, QObject *parent)
-: QThread(parent)
-, m_pFiffInfo(p_pFiffInfo)
-, m_bIsRunning(false)
-{
-    qRegisterMetaType<RTPROCESSINGLIB::FittingResult>("RTPROCESSINGLIB::FittingResult");
+void RtHPISWorker::doWork(const Eigen::MatrixXd& matData,
+            const Eigen::MatrixXd& m_matProjectors,
+            const QVector<int>& vFreqs,
+            QSharedPointer<FIFFLIB::FiffInfo> pFiffInfo) {
+
+    //Perform actual fitting
+    FittingResult fitResult;
+    fitResult.devHeadTrans.from = 1;
+    fitResult.devHeadTrans.to = 4;
+
+    HPIFit::fitHPI(matData,
+                    m_matProjectors,
+                    fitResult.devHeadTrans,
+                    vFreqs,
+                    fitResult.errorDistances,
+                    fitResult.fittedCoils,
+                    pFiffInfo);
+
+    emit resultReady(fitResult);
 }
 
 
 //*************************************************************************************************************
```

Comment:
Consider putting a more explicit command line here  - since a different class starts below:

  //============================================================================================================= 
 // DEFINE MEMBER METHODS OF RtHPIS
 //=============================================================================================================